### PR TITLE
Feature add test suite

### PR DESCRIPTION
**Pull request by rawrmonstar**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,7 +18,11 @@
   "homepage": "https://github.com/formidablelabs/karma-benchmarkjs-reporter",
   "scripts": {
     "preversion": "npm test",
-    "test": "eslint src"
+    "lint": "concurrently --raw --kill-others 'npm run lint-src' 'npm run lint-spec'",
```
Comment (ryan-roemer) [inline package.json]: Please switch to `builder concurrent <task1> <task2>`. You don't need to use archetypes, just the simple execution tool...

File: test/spec/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+describe("index", function () {
+  it("writes out formatted suite headings as they change");
```
Comment (ryan-roemer) [inline test/spec/index.spec.js:4]: Add ticket(s) and link to ticket in a code comment here to capture technical debt.

File: test/spec/sample-data.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
+var _ = require("lodash");
+var style = require("../../src/formatting/style");
+
+var getBenchConfig = function () {
+  return _.clone({
```
Comment (ryan-roemer) [inline test/spec/sample-data.js]: NIT: `_.clone()` unnecessary. It's _always_ a new object when this function is called (unless you're `cloneDeep` and mutating the `style` or something deep, which isn't the case here with shallow `clone`).

File: test/spec/sample-data.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    { name: "PhantomJS 2.1.1 (Mac OS X 0.0.0)" },
+    { name: "Chrome 52.0.2743.116 (Mac OS X 10.11.6)" },
+    { name: "Safari 9.1.3 (Mac OS X 10.11.6)" },
+    { name: "Firefox 48.0.2 (Mac OS X 10.11.6)" }
+  ];
+
+  qty = qty || 1;
+  qty = Math.min(qty, browsers.length);
+
+  return qty === 1 ? browsers[0] : browsers.slice(0, qty);
```
Comment (ryan-roemer) [inline test/spec/sample-data.js]: For simplicity, collapse to `return browsers.slice(0, qty);`

File: test/spec/formatting/format-suite-summary.spec.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+var chalk = require("chalk");
+var EOL = require("os").EOL;
+var formatSuiteSummary = require("../../../src/formatting/format-suite-summary");
+var sampleData = require("../sample-data");
+
+describe("formatSuiteSummary()", function () {
+  beforeEach(function () {
+    var NUM_BENCHMARKS = 6;
+    var NUM_BROWSERS = 2;
+    this.suite = sampleData.getSuite(NUM_BENCHMARKS, NUM_BROWSERS);
```
Comment (ryan-roemer) [inline test/spec/formatting/format-suite-summary.spec.js]: GLOBAL: Don't use `this.*` variables in any mocha  suites / tests. For large / complex suites, they play very poorly over time with Karma.

Instead, do scoped variables within the `describe` scope like:

``` js
describe("formatSuiteSummary()", function () {
  // Declare, but don't instantiate.
  var suite;
  var benchConfig;

  beforeEach(function () {
    var NUM_BENCHMARKS = 6;
    var NUM_BROWSERS = 2;
    // In the `beforeEach` actually set.
    suite = sampleData.getSuite(NUM_BENCHMARKS, NUM_BROWSERS);
    benchConfig = sampleData.getBenchConfig();
  });
```

File: test/spec/formatting/format-benchmark.spec.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+  it("should strip colors if colors set to false", function () {
+    this.benchConfig.colors = true;
+    var benchWithColor = formatBenchmark(this.benchmark, this.browser, this.benchConfig);
+    this.benchConfig.colors = false;
+    var benchWithoutColor = formatBenchmark(this.benchmark, this.browser, this.benchConfig);
+    expect(benchWithoutColor).to.equal(chalk.stripColor(benchWithColor));
+  });
+});
+
```
Comment (ryan-roemer) [inline test/spec/formatting/format-benchmark.spec.js]: NIT: Remove extra newline.

File: test/spec/formatting/format-benchmark.spec.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
+var expect = require("chai").expect;
+var chalk = require("chalk");
+var EOL = require("os").EOL;
+var formatBenchmark = require("../../../src/formatting/format-benchmark");
+var sampleData = require("../sample-data");
+
+describe("formatBenchmark()", function () {
+  beforeEach(function () {
+    this.benchmark = sampleData.getBenchmark(1);
```
Comment (ryan-roemer) [inline test/spec/formatting/format-benchmark.spec.js]: Same comment here.

**Review by ryan-roemer — CHANGES_REQUESTED** [review]
Nice work! First pass review done. Ping for re-review.

**Comment by rawrmonstar** [comment]
@ryan-roemer Thanks for the review! Your requested changes have been made.

**Review by ryan-roemer — APPROVED** [review]
LGTM!

**Pull request merged**